src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write code here
+const employeesArray = document.querySelectorAll('li');
+const employeesList = [];
+const ulElement = document.querySelector('ul');
+
+// Перебираем элементы и добавляем объекты в массив
+employeesArray.forEach((employee) => {
+  employeesList.push({
+    name: employee.textContent.trim(),
+    position: employee.dataset.position,
+    salary: Number(employee.dataset.salary.replace('$', '').replace(',', '')),
+    age: employee.dataset.age,
+  });
+});
+
+// Очищается список еллементов
+ulElement.innerHTML = '';
+
+employeesList
+  .sort((a, b) => b.salary - a.salary)
+  .forEach((employee) => {
+    const li = document.createElement('li');
+
+    li.textContent = `${employee.name}`;
```

Comment:
Currently, only the employee's name is being displayed in the list. Consider including additional information such as position, salary, or age to provide more context.